.github/workflows/tests.yml

```diff
@@ -247,141 +325,152 @@ jobs:
     # things to test. Running more tests is time consuming due to emulation
     # (factor 2-10x slower).
     runs-on: ubuntu-latest
+    defaults:
+      run:
+        # https://github.com/conda-incubator/setup-miniconda#use-a-default-shell
+        shell: bash -el {0}  # bash exit immediately on error + login shell
     strategy:
       fail-fast: false
       matrix:
-        default-channel: ['defaults', 'conda-forge']
         python-version: ['3.12']
-        platform: ['arm64', 'ppc64le']
-    env:
-      PYTHON: ${{ matrix.python-version }}
+        image: ['continuumio/miniconda3:latest', 'condaforge/miniforge3:latest']
+        platform: [arm64, ppc64le]
+
     steps:
-      - uses: actions/checkout@v4
+      - name: Checkout Source
+        uses: actions/checkout@v4
         with:
           fetch-depth: 0
 
-      - name: Set up QEMU
+      - name: Setup QEMU
         uses: docker/setup-qemu-action@v3
-        # Equivalent to locally run:
-        #   `docker run --privileged --rm tonistiigi/binfmt --install all`
 
-      - name: Python linux-${{ matrix.platform }} on ${{ matrix.default-channel }}, ${{ matrix.python-version }} tests
-        run: >
-          docker run
-          --rm -v ${PWD}:/opt/conda-src
+      - name: Run Tests
+        run: docker run
+          --rm
+          --volume ${PWD}:/opt/conda-src
+          --workdir /opt/conda-src
           --platform linux/${{ matrix.platform }}
-          ghcr.io/conda/conda-ci:main-linux-python${{ matrix.python-version }}${{ matrix.default-channel == 'conda-forge' && '-conda-forge' || '' }}
-          /opt/conda-src/dev/linux/qemu.sh
-
-      - name: Upload test results
-        if: '!cancelled()'
-        uses: actions/upload-artifact@v4
-        with:
-          # name has to be unique, to not overwrite uploads of other matrix runs. sha1 is optional and only to differentiate
-          # when locally dowloading and comparing results of different workflow runs.
-          name: test-results-${{ github.sha }}-linux-${{ matrix.platform }}-qemu-${{ matrix.default-channel }}-${{ matrix.python-version }}
-          path: |
-            .coverage
-            test-report.xml
-          retention-days: 1
+          ${{ matrix.image }}
+          bash -lc
+          ". /opt/conda/etc/profile.d/conda.sh
+          && conda create
```

Comment:
```suggestion
          && set -x
          && conda create
```

Reply:
and we don't want to do `bash -lcx ...`?